dataprofiler/profilers/profile_builder.py

```diff
@@ -806,15 +809,35 @@ def load(cls, filepath):
             UnstructuredProfiler
         :rtype: BaseProfiler
         """
-        profile_options = cls._option_class()
-        profile_options.data_labeler.is_enabled = False
-        profiler = cls(None, options=profile_options)
-
         # Load profile from disk
         with open(filepath, "rb") as infile:
             data = pickle.load(infile)
-            for key in data:
-                setattr(profiler, key, data[key])
+
+        # remove profiler class if it exists
+        profiler_class = data.pop('profiler_class', None)
+
+        # if the user didn't load from the a given profiler class, we need
+        # to determine which profiler is being loaded.
+        profiler_cls = cls
+        if cls is BaseProfiler:
+            if 'StructuredProfiler' == profiler_class:
+                profiler_cls = StructuredProfiler
+            elif 'UnstructuredProfiler' == profiler_class:
```

Comment:
Could you flip these equality statements? Feels weird to see <value> == <var> rather than <var> == <value>

Reply:
done